examples/taskruns/workspace-readonly.yaml

```diff
@@ -0,0 +1,42 @@
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: my-pvc-3
+spec:
+  resources:
+    requests:
+      storage: 5Gi
+  volumeMode: Filesystem
+  accessModes:
+    - ReadWriteOnce
+---
+apiVersion: tekton.dev/v1alpha1
+kind: TaskRun
+metadata:
+  generateName: workspaces-readonly-
+spec:
+  workspaces:
+    - name: write-allowed
+      persistentVolumeClaim:
+        claimName: my-pvc-3
+    - name: write-disallowed
+      persistentVolumeClaim:
+        claimName: my-pvc-3
+  taskSpec:
+    workspaces:
+    - name: write-allowed
+    - name: write-disallowed
+      readOnly: true
+    steps:
+    - name: write-allowed
+      image: ubuntu
+      script: echo "hello" > $(workspaces.write-allowed.path)/foo
+    - name: read-allowed
+      image: ubuntu
+      script: cat $(workspaces.write-allowed.path)/foo | grep "hello"
+    - name: write-disallowed
+      image: ubuntu
+      script: echo "goodbye" > $(workspaces.write-disallowed.path)/foo || true
```

Comment:
Nice!

Reply:
`|| touch write-failed.txt` then checking if the file exists would give you a stronger signal that the write operation failed, and didn't just succeed to write and then the contents were lost.

Reply:
good call.